Explain how to activate, deactivate & uninstall one or more blocks (#29)
retraceur/retraceur.github.io: src/content/docs/administration/manage-blocks.mdx

---
title: Manage Blocks

> description: This guide is explaining how to install, activate, update, deactivate and delete Retraceur Blocks.

Author: ```suggestion
description: This guide is explaining how to manually install & update, activate, deactivate and uninstall Retraceur Blocks.
```

### Manual Installation

When the Block's directory does not yet exist on your Website, Retraceur will create a new directory for it inside your Website `/wp-content/plugins` parent directory. Then, Retraceur will tell you whether the process was successful or not.

![Install a Block](../../../assets/images/manage-blocks-v400-us-003.png)

### Manual Upgrade (or downgrade)

When the Block's directory already exists on your Website, a confirmation screen will ask you whether you want to overwrite it with the new uploaded Block.

![Up/Downgrade a Block](../../../assets/images/manage-blocks-v400-us-004.png)

Check the displayed information and specifically the "Version" field to make sure you're doing what you intend to do. If that's the case, click on the "Replace current with uploaded" button otherwise click on the "Cancel and go back" one. If you chose to proceed, a screen will inform you about the installation/update status. Once complete, you'll find your Block inside the list displayed into the "Installed Blocks" administration page.

## Activating Blocks

> As shown in the screenshot from the [Manual Installation](#manual-installation) subsection, the confirmation message for installing a block includes an “Activate” link to integrate the block into Retraceur's page-loading process for your website. You can also activate one or more blocks in two other ways.

Author: ```suggestion
As shown in the screenshot from the [Manual Installation](#manual-installation) subsection, the confirmation message for installing a Block includes an “Activate” link to integrate the Block into Retraceur's page-loading process for your website. You can also activate one or more Blocks in two other ways.
```

### Block-by-Block Activation

> Below each block's name, the first blue link activates the corresponding block.

Author: ```suggestion
Below each Block's name, the first blue link activates the corresponding Block.
```

![Block activation](../../../assets/images/manage-blocks-activate-us-001.png)

> After clicking this link, the page will reload and confirm that the operation was successful with a message on a green background displayed above the table. Additionally, the row in the table corresponding to any activated block will have a thicker, dark blue left border and a light blue background.

Author: ```suggestion
After clicking this link, the page will reload and confirm that the operation was successful with a message on a green background displayed above the table. Additionally, the row in the table corresponding to any activated Block will have a thicker, dark blue left border and a light blue background.
```

![Block activated](../../../assets/images/manage-blocks-activate-us-002.png)

### Bulk Activation

> You can also choose to activate one or more blocks by selecting the corresponding checkboxes before using one of the two drop-down lists under “Batch Actions” to define the activation action, as shown below.

Author: ```suggestion
You can also choose to activate one or more Blocks by selecting the corresponding checkboxes before using one of the two drop-down lists under “Batch Actions” to define the activation action, as shown below.
```

![Blocks activation](../../../assets/images/manage-blocks-activate-us-003.png)

> Next, click the “Apply” button to the right of the drop-down list to start the process. When the page reloads, you’ll see that the selected block(s) have been activated, as indicated by a message with a green background positioned just above the block table.

Author: ```suggestion
Next, click the “Apply” button to the right of the drop-down list to start the process. When the page reloads, you’ll see that the selected Block(s) have been activated, as indicated by a message with a green background positioned just above the Block table.
```

![Blocks activated](../../../assets/images/manage-blocks-activate-us-004.png)

## Deactivating Blocks

> You can deactivate one or more blocks in two ways.

Author: ```suggestion
You can deactivate one or more Blocks in two ways.
```

### Deactivating Blocks One by One

> Below each block’s name, the only available link lets you deactivate that block.

Author: ```suggestion
Below each Block’s name, the only available link lets you deactivate that Block.
```

![Deactivate a Block](../../../assets/images/manage-blocks-deactivate-us-001.png)

> After clicking this link, the page will reload and confirm that the operation was successful with a message on a green background displayed above the table. Additionally, the row in the table for any deactivated block will not have a thicker border, and its background will be transparent.

Author: ```suggestion
After clicking this link, the page will reload and confirm that the operation was successful with a message on a green background displayed above the table. Additionally, the row in the table for any deactivated Block will not have a thicker border, and its background will be transparent.
```

![Deactivated block](../../../assets/images/manage-blocks-deactivate-us-002.png)

### Batch Deactivation

> You can also choose to deactivate one or more blocks by checking the corresponding checkboxes before using one of the two drop-down lists under “Batch Actions” to select the deactivation action, as shown below.

Author: ```suggestion
You can also choose to deactivate one or more Blocks by checking the corresponding checkboxes before using one of the two drop-down lists under “Batch Actions” to select the deactivation action, as shown below.
```

![Deactivate blocks](../../../assets/images/manage-blocks-deactivate-us-003.png)

> Then click the “Apply” button just to the right of the drop-down list to start the operation. When the page reloads, you’ll see that the selected block(s) have indeed been deactivated, as indicated by a message with a green background positioned just above the block table.

Author: ```suggestion
Then click the “Apply” button just to the right of the drop-down list to start the operation. When the page reloads, you’ll see that the selected Block(s) have indeed been deactivated, as indicated by a message with a green background positioned just above the Block table.
```

![Deactivated blocks](../../../assets/images/manage-blocks-deactivate-us-004.png)

## Uninstalling Blocks

> <Aside type="note">As long as a block is active, you cannot delete (uninstall) it. Be sure to deactivate it before uninstalling it</Aside>

Author: ```suggestion
<Aside type="note">As long as a Block is active, you cannot delete (uninstall) it. Be sure to deactivate it before uninstalling it.</Aside>
```

> You can uninstall one or more blocks in two ways.

Author: ```suggestion
You can uninstall one or more Blocks in two ways.
```

### Uninstalling one block at a time

> Below each block’s name, the second red link allows you to delete the corresponding block.

Author: ```suggestion
Below each Block’s name, the second red link allows you to delete the corresponding Block.
```

![Delete a block](../../../assets/images/manage-blocks-delete-us-001.png)

After clicking this link, a dialog box will ask you to confirm your action.

![Deletion Confirmation](../../../assets/images/manage-blocks-delete-us-002.png)

> After confirmation, the row for the corresponding block will be replaced by a message confirming that the block has been deleted.

Author: ```suggestion
After confirmation, the row for the corresponding Block will be replaced by a message confirming that the Block has been deleted.
```

### Bulk Uninstallation

> You can also choose to uninstall one or more blocks by checking the corresponding checkboxes before using one of the two drop-down lists under “Bulk Actions” to select the delete action, as shown below.

Author: ```suggestion
You can also choose to uninstall one or more Blocks by checking the corresponding checkboxes before using one of the two drop-down lists under “Bulk Actions” to select the delete action, as shown below.
```

![Delete blocks](../../../assets/images/manage-blocks-delete-us-003.png)

> Then click the “Apply” button immediately to the right of the drop-down list. An intermediate page will ask you to confirm that you want to delete the selected block(s). Once confirmed, the block(s) will be uninstalled.

Author: ```suggestion
Then click the “Apply” button immediately to the right of the drop-down list. An intermediate page will ask you to confirm that you want to delete the selected Block(s). Once confirmed, the Block(s) will be uninstalled.
```
